### PR TITLE
Add prompt muons CORSIKA EHISTORY simulation

### PR DESCRIPTION
**Pull request by pascalgutjahr**
Add chain to simulate CORSIKA 77420 using the extended history option to study atmospheric prompt muons and neutrinos. Also one example config is added.

### CODE REVIEW

File: configs/prompt_muons_example.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    'trimshower': True,
+    'pnorm': [10., 5., 3., 2., 1.],
+    'pgam': [1., 1., 1., 1., 1.],
+    'UsePipe': True,
+    
+    'ecuts1': 273.,  # hadron min energy (see corsika docs)
+    'ecuts2': 273.,  # muon min energy (see corsika docs)
+    'ecuts3': 273.,  # electron min energy (see corsika docs)
+    'ecuts4': 273.,  # photon min energy (see corsika docs)
+    'CVMFS': '/data/user/pgutjahr/software/CORSIKA/',
```
Comment (mhuen) [inline configs/prompt_muons_example.yaml]: Needs to be moved to CVMFS

File: configs/prompt_muons_example.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    # things that should be defined by user:
+    'polyplopia': True,
+    'mctree_polyplopia': 'I3MCTree',
+    'eprimarymin': !!float 1e8,
+    'eprimarymax': !!float 1e9,
+    'nuaddi': True,
+}
+muon_propagation_config : {
+    'PROPOSAL_config_file': '/data/user/pgutjahr/software/icetray/v1.5.0/src/PROPOSAL/resources/config_icesim.json',
```
Comment (mhuen) [inline configs/prompt_muons_example.yaml]: is a custom config required? It seems to be the default config by PROPOSAL. Todo: check if this is used and utilize official icetray one

File: job_templates/py3-v4.2.1.sh [hunk 3]
```diff
@@ -13,14 +13,16 @@ FINAL_OUT={final_out}
 KEEP_CRASHED_FILES={keep_crashed_files}
 
 
+cd
+echo 'PWD: '$(pwd)
 echo 'Starting job on Host: '$HOSTNAME
 echo 'Loading py3-v4.2.1'
 eval `/cvmfs/icecube.opensciencegrid.org/py3-v4.2.1/setup.sh`
-export PYTHONUSERBASE=/INSERT/PATH/HERE
+export PYTHONUSERBASE=/data/user/lneste/prompt_muons/py3-v4.2.1
```
Comment (mhuen) [inline job_templates/py3-v4.2.1.sh]: Let's move these to official releases if possible / do we need any additional packages? Which ones and why?

File: job_templates/py3-v4.2.1.sh [hunk 4]
```diff
@@ -13,14 +13,16 @@ FINAL_OUT={final_out}
...
 echo 'Using PYTHONUSERBASE: '${PYTHONUSERBASE}
 
 export PATH=$PYTHONUSERBASE/bin:$PATH
-export PYTHONPATH=$PYTHONUSERBASE/lib/python3.7/site-packages:$PYTHONPATH
+export PYTHONPATH=$PYTHONUSERBASE/lib/python3.10/site-packages:$PYTHONPATH
```
Comment (mhuen) [inline job_templates/py3-v4.2.1.sh]: Maybe we should figure out a more robust way here that reduces places in which modifications need to be made when changing versions

File: steps/step_0_corsika_py3_v4_1_0_branch_TrimShower.py [hunk 5]
```diff
@@ -147,6 +150,18 @@ def main(cfg, run_number, scratch):
             **cfg['BaseSimulationBiasSettings']
         )
 
+
+    # -----------------------------------
+    # Delete keys
```
Comment (mhuen) [inline steps/step_0_corsika_py3_v4_1_0_branch_TrimShower.py]: we can make this more general and flexible by simply adding a list of keys to delete.

File: steps/step_0_corsika_py3_v4_2_1_prompt_muons_v1_5_1.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh /cvmfs/icecube.opensciencegrid.org/py3-v4.2.1/icetray-start
```
Comment (mhuen) [inline steps/step_0_corsika_py3_v4_2_1_prompt_muons_v1_5_1.py]: We should base this off of official releases whenever possible.

File: steps/step_2_snowstorm_detector_simulation_py3_v4_2_1_icetray_v1_5_1.py [hunk 7]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh /cvmfs/icecube.opensciencegrid.org/py3-v4.2.1/icetray-start
+#METAPROJECT /data/user/pgutjahr/software/icetray/v1.5.1/build
```
Comment (mhuen) [inline steps/step_2_snowstorm_detector_simulation_py3_v4_2_1_icetray_v1_5_1.py]: Should also update this meta project

File: steps/step_3_pass2_L1_py3_v4_2_1.py [hunk 8]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh /cvmfs/icecube.opensciencegrid.org/py3-v4.2.1/icetray-start
+#METAPROJECT /data/user/lneste/i3/icetray/build
```
Comment (mhuen) [inline steps/step_3_pass2_L1_py3_v4_2_1.py]: here as well

File: steps/step_4_pass2_L2_py3_v4_2_1.py [hunk 9]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh /cvmfs/icecube.opensciencegrid.org/py3-v4.2.1/icetray-start
+#METAPROJECT /data/user/lneste/i3/icetray/build
```
Comment (mhuen) [inline steps/step_4_pass2_L2_py3_v4_2_1.py]: same here

File: configs/prompt_muons_example.yaml
Comment (mhuen) [inline configs/prompt_muons_example.yaml:1]: is this just an example config? In that case I think we can remove it. We just need the default_* configs for default configuration settings that will be used if a user does not overwrite a specific one

Comment (mhuen) [reply configs/prompt_muons_example.yaml:1]: Moved to sub-directory with all other example configs that are not used as sources of default parameters in the `processing_chains`.

File: configs/prompt_muons_example.yaml (hunk 1 above)
Comment (pascalgutjahr) [reply configs/prompt_muons_example.yaml]: Now it is available at `/net/cvmfs_users/pgutjahr/software/CORSIKA/`. Also a README how to build CORSIKA is added.

Comment (pascalgutjahr) [reply configs/prompt_muons_example.yaml]: @The-Ludwig, can you adjust the `CVMFS` path? Thanks

Comment (The-Ludwig) [reply configs/prompt_muons_example.yaml]: @pascalgutjahr Of course, but why don't you just change that yourself? Am I missing a point here?

File: steps/step_0_corsika_py3_v4_2_1_prompt_muons_v1_5_1.py [hunk 10]
```diff
@@ -0,0 +1,6 @@
...
+#METAPROJECT /data/user/pgutjahr/software/icetray/v1.5.1/build
```
Comment (mhuen) [inline steps/step_0_corsika_py3_v4_2_1_prompt_muons_v1_5_1.py]: move to official icetray release if possible

**Pull request merged**